service/service.go

```diff
@@ -503,17 +504,29 @@ func (s *service) DeleteProductById(ctx context.Context, userId uuid.UUID, id uu
 	return err
 }
 
-func (s *service) GetProductsByStoreId(ctx context.Context, page int32, limit int32, storeID uuid.UUID, productTypeId *int64, isIncludeDeactivated bool) (products []*prodEntity.Product, pagination base_model.Pagination, err error) {
+func (s *service) GetProductsByStoreId(ctx context.Context, params types.GetProductsByStoreIdParams) (products []*prodEntity.Product, pagination base_model.Pagination, err error) {
 	pagination = base_model.Pagination{
-		Page:  page,
-		Limit: limit,
+		Page:  params.Page,
+		Limit: params.Limit,
 	}
 
-	if _, err := s.storeRepository.GetStore(ctx, storeID.String()); err != nil {
+	if _, err := s.storeRepository.GetStore(ctx, params.StoreID.String()); err != nil {
 		return nil, pagination, err
 	}
 
-	if products, pagination, err = s.productRepository.GetProductsByStoreId(ctx, pagination, storeID, productTypeId, isIncludeDeactivated); err != nil {
+	getProductsByStoreIdRepoParams := types.GetProductsByStoreIdRepoParams{
+		Pagination:           pagination,
+		StoreID:              params.StoreID,
+		ProductTypeId:        params.ProductTypeId,
+		IsIncludeDeactivated: params.IsIncludeDeactivated,
+		OrderBy:              params.OrderBy,
+		OrderAscDesc:         params.OrderAscDesc,
+	}
+
+	if products, pagination, err = s.productRepository.GetProductsByStoreId(ctx, getProductsByStoreIdRepoParams); err != nil {
+		if strings.Contains(err.Error(), ErrIncorrectSqlSyntax) || strings.Contains(err.Error(), ErrInvalidColumnName) {

```

Comment:
could we move all code related to DB error in the repository so the service does not have anything related to specific DB?

Reply:
Okay noted mas, will move the DB error to the repository